packages/go_router/lib/src/delegate.dart

```diff
@@ -67,15 +75,28 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
 
     _matches.push(newPageKeyMatch);
     notifyListeners();
+    return completer.future;
   }
 
   /// Returns `true` if there is more than 1 page on the stack.
   bool canPop() {
     return _matches.canPop();
   }
 
-  /// Pop the top page off the GoRouter's page stack.
-  void pop() {
+  /// Pop the top page off the GoRouter's page stack and complete a promise if
+  /// there is one.
+  void pop<T extends Object?>([T? value]) {
```

Comment:
Why is this the only method that returns `void`?

Reply:
Because the only thing that the .pop() does is complete the completer that the other screen is waiting to complete. So no asynchronous methods here.